Source/Core/VideoBackends/Metal/MTLMain.mm

```diff
@@ -119,8 +119,13 @@ static bool WindowSystemTypeSupportsMetal(WindowSystemType type)
 
     MRCOwned<CAMetalLayer*> layer = MRCRetain(static_cast<CAMetalLayer*>(wsi.render_surface));
     [layer setDevice:adapter];
-    if (Util::ToAbstract([layer pixelFormat]) == AbstractTextureFormat::Undefined)
-      [layer setPixelFormat:MTLPixelFormatBGRA8Unorm];
+    [layer setWantsExtendedDynamicRangeContent:YES];
+    [layer setPixelFormat:MTLPixelFormatRGBA16Float];
+
+    const CFStringRef name = kCGColorSpaceExtendedLinearSRGB;
+    CGColorSpaceRef colorspace = CGColorSpaceCreateWithName(name);
+    [layer setColorspace:colorspace];
+    CGColorSpaceRelease(colorspace);
```

Comment:
SDR mode is noticeably different with this change (left is before this change, right is after):
<img width="1425" alt="image" src="https://github.com/dolphin-emu/dolphin/assets/3315070/33a25532-eb96-4398-9aff-bec1d160b685">

Also, using RGBA16 disables full screen optimizations on my Intel mac (`MTL_HUD_ENABLED=1`, watch for whether it says composited or direct when fullscreen).  You probably only want to do this if the user requested HDR.